### ISSUE TITLE
Pronouns on name tag

### ISSUE THREAD

**Issue by MargoSizemore** [Yellow-Dog-Man/Resonite-Issues#213]
### Is your feature request related to a problem? Please describe.

Now that people have the option to hide custom name tags, people's pronouns on their name tags may not be seen by some people. People that have bad intentions can also act like they did not by just saying "I didn't see it."

### Describe the solution you'd like

Ability to add pronouns to the default name tag, or adding a badge for pronouns in a similar way to badges for being hard of hearing or mute through the Resonite bot.

### Describe alternatives you've considered

I could make it always show no matter what, but that would be obtrusive, and I believe there are more elegant solutions. I could make it a physical thing on my avatar, but then people off to the side of me who look over at me while talking about me may not see it.

**Comment by Pinkyyyy** [comment]
I agree, adding an avatar agnostic, similar to how the mute badge works, pronoun name tag, or just make a new thing for the game: pronoun tag, that can be easily set in the dashboard would be amazing.

Additionally: better tutorialization on how to get system wide badges/name tags etc. would be great. If a user didn't tell me i couldn't have known that's even a thing, or that you have to use the Resonite bot in game, idk even what commands that thing has!

**Comment by MargoSizemore** [comment]
> I agree, adding an avatar agnostic, similar to how the mute badge works, pronoun name tag, or just make a new thing for the game: pronoun tag, that can be easily set in the dashboard would be amazing.
> 
> 
> 
> Additionally: better tutorialization on how to get system wide badges/name tags etc. would be great. If a user didn't tell me i couldn't have known that's even a thing, or that you have to use the Resonite bot in game, idk even what commands that thing has!

There's going to be a wiki eventually that should say all the commands you may need.

**Comment by Zeriga** [comment]
Pronoun field or system badge would be great.

**Comment by Nytra** [comment]
This might be something better suited for a user profile system. You could click the nametag to be taken to that user's profile where you can see pronouns, badges, bio. etc

**Comment by The-furf-of-July** [comment]
> This might be something better suited for a user profile system. You could click the nametag to be taken to that user's profile where you can see pronouns, badges, bio. etc

Some people and some cases are sensitive enough to warrant the information being unavoidable.

Like the original post said, 
> People that have bad intentions can also act like they did not by just saying "I didn't see it."

> people off to the side of me who look over at me while talking about me may not see it.

And there's still the case of plurals, where they straight up need to display different names at different times. A subtitle would be a better fix than relying on people checking your profile.

**Comment by Zyzyl** [comment]
In the original post, you didn't specify what visibility level any additional pronouns etc. should have @MargoSizemore. I assume you mean for them to behave the same way as the default nameplate?

If so, this _would_ make your pronouns visible in cases where custom nameplates are turned off. However, it's possible to turn off the default nameplates as well. Thus people would have some plausible deniability for having not seen your pronouns even were this implemented. As such, would the suggested solution still solve the problem to your satisfaction?

The ability to hide default nameplates needs to be preserved because there are contexts where it's important to have all floating text and badges hidden (e.g. where immersion is paramount like the MetaMovie).

**Comment by MargoSizemore** [comment]
If someone has all name tags turned off then there's not much I can do. I have boobs parented onto my avatar for now until I have a custom one, I was hoping it would help make people misgender me less, but it didn't really seem to help. I've changed my username to have my pronouns in it for now until my Patreon goes through to set that as my custom badge, if someone ignores that and ignores my name and badge once I have that, then there's nothing I can really do other than report them to moderation if I correct them and they continue, and I believe it is not on accident.

**Comment by Frooxius** [comment]
I was thinking of adding pronouns as part of the user profile once we implement that, along with a number of other things. We'll figure out a good way to display it too.

If I understand it correctly, you also wish these to be displayed as part of default nameplates when configured?

**Comment by MargoSizemore** [comment]
Yup! For now, having my pronouns in my display name works for me.

**Comment by Zilepo** [comment]
Just want to concur and say that once the default nameplate is customized, it would be good to have the ability to include pronouns as part of it

**Comment by shiftyscales** [comment]
Make sure to [react with a thumb up](https://github.com/Yellow-Dog-Man/Resonite-Issues/blob/main/HOW_TO_UPVOTE.md) to the original post on any issues you wish to signal interest in being prioritized, @Zilepo.

**Comment by NukiRaccoon** [comment]
Just a thought, if more information is added to people's nametags: my own avatar features a system which causes the name tag to fold out to display more information like that, such as my pronouns and badges.  Perhaps a system like this could be implemented by default, which reacts when people are looking directly at it.

**Comment by Frooxius** [comment]
That'd be a separate feature for name tags, which should be it's own issue if you want that added. We'll probably rework the default nameplates at some point using templates, but that is not dependent on this issue.

**Comment by shiftyscales** [comment]
Created #3820 as a hub for tracking nameplate-related issues / making the NamePlateInterface customizable to the art team / users.

**Comment by eth0fox** [comment]
Maybe we could expand the existing default Cloud Variable system to allow for customisation like this? Just like how avatars can currently customise their colour scheme based on who's wearing them using the `G-Resonite.UserSettings.Color.(Primary|Secondary|Tertiary)` variables, we could possibly have things like `G-Resonite.UserSettings.Pronouns`, or `G-Resonite.UserSettings.DisplayName`?

**Comment by Frooxius** [comment]
No,  we would not use cloud variables, that would just make it more complicated. This would be more inherent part of the system.

**Comment by Frooxius** [comment]
This has been added in 2026.6.1.1293!

**Comment by lynrayy** [comment]
woke enough

**Comment by Frooxius** [comment]
It's what we gotta do :3

Too many sleep on Resonite!

**Comment by SizemoreSystem** [comment]
I believe the validation rules should be changed to allow "ask", "ask us", and "ask me".

**Comment by eth0fox** [comment]
> I believe the validation rules should be changed to allow "ask", "ask us", and "ask me".

mhhh- I think having the pronouns field be structured is nice (ie maybe something could read it and automatically say "Ask **him** for xyz", and allowing freeform text would become "Ask **ask me** for xyz" which doesn't make sense)

For the case of just string the field to "ask me", does just leaving the field blank not already imply that?

**Comment by SizemoreSystem** [comment]
> > I believe the validation rules should be changed to allow "ask", "ask us", and "ask me".
> 
> mhhh- I think having the pronouns field be structured is nice (ie maybe something could read it and automatically say "Ask **him** for xyz", and allowing freeform text would become "Ask **ask me** for xyz" which doesn't make sense)
> 
> For the case of just string the field to "ask me", does just leaving the field blank not already imply that?

wouldn't that require the format "they/them/their/theirs/themselves"?
